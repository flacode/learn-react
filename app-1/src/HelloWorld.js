import React, { Component }  from 'react';
import './HelloWorld.css';

class HelloWorld extends Component{

    constructor(props){
        super(props); // constructor for parent
        this.state = {greeting: 'Hello'};
        this.frenchify = this.frenchify.bind(this);// bind the event to an instance This tells Javascript “Hey, any time you see a reference to this inside of the frenchify function, I want you specifically to refer to ME.” 
        this.removeGreeting = this.removeGreeting.bind(this);
    }

    frenchify(){
        this.setState({greeting: 'Bonjour'});
    }

    removeGreeting(){
        this.props.removeGreeting(this.props.name);
    }

    render(){
        return (
        <div className="HelloWorld">
            {this.state.greeting} { this.props.name }
            <br/>
            <button className="frenchify" onClick={ this.frenchify }>Frenchify!</button>
            <button className="remove" onClick={this.removeGreeting}>Remove Me!</button>
            </div>
        );
    }
}

export default HelloWorld;