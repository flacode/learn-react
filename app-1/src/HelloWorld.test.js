import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer'; // we start off by calling renderer.create, and passing to that function the JSX that we want to build into our new snapshot. 
import HelloWorld from './HelloWorld';

configure({adapter: new Adapter() });

describe(HelloWorld, () => {
    // Add the rest of our tests here later!
    const name = 'Person';
    const mockRemoveGreeting = jest.fn();//jest.fn() is a special syntax to create a mock function. 
    const component = shallow(
      <HelloWorld name={name} removeGreeting={mockRemoveGreeting}/>
    );

    it('renders and matches our snapshot', () => {
        const component = renderer.create(
          <HelloWorld name="Person"/>
        );
        const tree = component.toJSON();//take that snapshot and turn it into a JSON representation of the tree
        expect(tree).toMatchSnapshot();
      });

      it('contains the supplied name', () => {
        expect(component.text()).toContain(name);
      });

      it('modifies the greeting when frenchify button is clicked', () =>{
          component.find('button.frenchify').simulate('click');
          expect(component.text()).toContain('Bonjour');
      });

      it('calls the passed in removeGreeting function when remove button is clicked', () => {
          component.find('button.remove').simulate('click');
          expect(mockRemoveGreeting).toBeCalled();
      });
      

  });