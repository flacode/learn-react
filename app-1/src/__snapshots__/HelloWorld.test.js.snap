exports[`class HelloWorld extends /* istanbul ignore next */_react.Component {

    constructor(props) {
        super(props); // constructor for parent
        this.state = { greeting: 'Hello' };
        this.frenchify = this.frenchify.bind(this); // bind the event to an instance This tells Javascript “Hey, any time you see a reference to this inside of the frenchify function, I want you specifically to refer to ME.” 
        this.removeGreeting = this.removeGreeting.bind(this);
    }

    frenchify() {
        this.setState({ greeting: 'Bonjour' });
    }

    removeGreeting() {
        this.props.removeGreeting(this.props.name);
    }

    render() {
        return (
            /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'div', /* istanbul ignore next */{ className: 'HelloWorld', __source: { fileName: _jsxFileName, lineNumber: 23 }, __self: this },
                this.state.greeting, /* istanbul ignore next */' ', this.props.name,
                /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'br', /* istanbul ignore next */{ __source: { fileName: _jsxFileName, lineNumber: 25 }, __self: this }),
                /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'button', /* istanbul ignore next */{ className: 'frenchify', onClick: this.frenchify, __source: { fileName: _jsxFileName, lineNumber: 26 }, __self: this }, /* istanbul ignore next */'Frenchify!'),
                /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'button', /* istanbul ignore next */{ className: 'remove', onClick: this.removeGreeting, __source: { fileName: _jsxFileName, lineNumber: 27 }, __self: this }, /* istanbul ignore next */'Remove Me!')));


    }} renders and matches our snapshot 1`] = `
<div
  className="HelloWorld">
  Hello
   
  Person
  <br />
  <button
    className="frenchify"
    onClick={[Function]}>
    Frenchify!
  </button>
  <button
    className="remove"
    onClick={[Function]}>
    Remove Me!
  </button>
</div>
`;
